### PR TITLE
Clojure 1.11 refactoring

### PR DESCRIPTION
**Pull request by kelvinqian00**
Use features introduced in Clojure 1.11, such as kwarg map passing, `update-vals`, `:as-alias`, and new parse functions.

### CODE REVIEW

File: src/dev/series.clj [hunk 1]
```diff
@@ -1,5 +1,5 @@
 (ns series
-  (:require [java-time       :as t]
+  (:require [java-time.api   :as t]
```
Comment (milt) [inline src/dev/series.clj:2]: I'm getting deja vu, thought we already did this?

File: src/main/com/yetanalytics/datasim.clj [hunk 2]
```diff
@@ -15,29 +15,31 @@
...
 ;; TODO: Extract more implementation code from `sim` namespace
-;; TODO: Use more opt-map passing (instead of `apply`) in Clojure 1.11
 
+#_{:clj-kondo/ignore [:unused-binding]}
 (defn generate-seq
   "Given `input`, produce a lazy sequence of statements in a synchronous
    fashion."
-  [input & kwargs]
-  (apply sim/sim-seq input kwargs))
+  [input & {:keys [select-agents] :as kwargs}]
```
Comment (milt) [inline src/main/com/yetanalytics/datasim.clj:23]: So the unused bindings are basically for documentation purposes? I don't hate that.

File: src/dev/series.clj (hunk 1 above)
Comment (kelvinqian00) [reply src/dev/series.clj:2]: I thought so too, but I guess I missed a spot.

File: src/main/com/yetanalytics/datasim.clj (hunk 2 above)
Comment (kelvinqian00) [reply src/main/com/yetanalytics/datasim.clj:23]: Yep, which is pretty important for the top-level API functions.

**Review by milt — APPROVED** [review]
Looks good. `update-vals` does save some code.

**Pull request merged**